src/text_to_binary/rewrite.ml

```diff
@@ -407,6 +407,131 @@ let rewrite_types (_modul : Assigned.t) (t : binary str_type) :
   let t = [ (None, (Final, [], t)) ] in
   Ok t
 
+let rec rewrite_term (binder_list : string option list)
+  ?(modul : Binary.modul = Binary.empty_modul)
+  ?(func_param_list : string option list = []) :
```

Comment:
Could you use named parameters instead of optional ones here ? This piece of code is already complicated and it's too easy to do a mistake with optional parameters IMO...